### ISSUE TITLE
Allow other component names than the build-in ones

### ISSUE THREAD

**Issue by atimmer** [formatjs/babel-plugin-react-intl#138]
`COMPONENT_NAMES` only allows `FormattedMessage` and `FormattedHTMLMessage`. I would like an option to the babel plugin that allows adding components to this list.

The reason is that we have a `FormattedScreenReaderMessage` component that renders a message as screen reader text. `ScreenReaderText` is a component that only accepts a `string` as its children. This is because screen reader text should never be shown visually. Opening up `ScreenReaderText` to arbitrary components opens up the possibility that someone puts a block element in there, breaking this 'promise'.

The `FormattedScreenReaderMessage` code looks like this:

```jsx
// External dependencies.
import React from "react";
import { FormattedMessage } from "react-intl";
import omit from "lodash/omit";
import PropTypes from "prop-types";

// External dependencies.
import ScreenReaderText from "./ScreenReaderText";

/**
 * Combines the text that is passed to the formatted screen reader message.
 *
 * @param {string[]} textNodes The text nodes passed from the FormattedMessage
 *                             component.
 * @param {Object}   props     The props to the `FormattedScreenReaderMessage`
 *                             component.
 * @returns {string} The combined text.
 */
function renderText( textNodes, props ) {
	const combinedParts = [ props.before, ...textNodes, props.after ];

	return combinedParts.join( "" );
}

/**
 * Renders a message which should only be shown to screen readers. All props
 * will be passed for `FormattedMessage` except `before` and `after`.
 *
 * With `before` and `after` a string can be put before or after the translated
 * string. This is useful for punctuation that shouldn't be translated. Such
 * punctuation can for example be a colon.
 *
 * @param {Object} props        The view properties.
 * @param {string} props.before A piece of text to render before the translation.
 * @param {string} props.after  A piece of text to render after the translation.
 *
 * @returns {ReactElement} The rendered div containing the screen reader text.
 */
const FormattedScreenReaderMessage = ( props ) => {
	return (
		<FormattedMessage { ...omit( props, "children" ) }>
			{ ( textNodes ) => <ScreenReaderText>{ renderText( textNodes, props ) }</ScreenReaderText> }
		</FormattedMessage>
	);
};

FormattedScreenReaderMessage.propTypes = {
	before: PropTypes.string,
	after: PropTypes.string,
	values: PropTypes.objectOf( PropTypes.string ),
};

export default FormattedScreenReaderMessage;
```

So `FormattedScreenReaderMessage` accepts all the arguments `FormattedMessage` accepts but it wraps the translated message to be screen reader text.

By allowing `COMPONENT_NAMES` to be expanded using an option I could add `FormattedScreenReaderMessage` to the array of accepted components and our code can make use of the component.

**Comment by martinratinaud** [comment]
I agree,

Also if we want to abstract the use of react-intl by using a more general component <Translate> with the same signature, we should be able to extract the corresponding data

:-)

**Comment by sanbornhilland** [comment]
What's the status of this issue and #140? I was thinking of writing this myself but if I can contribute to the official effort I'd rather do that.

**Comment by dan-lee** [comment]
@sanbornhilland I'm also in need of this.
I think the linked PR tries to tackle too many things at once. Only expanding `COMPONENT_NAMES` would be a good first step. I would also be willing to to contribute.

**Comment by SleepWalker** [comment]
It's not that hard to split it into parts. The problem, that this repo seems to be not maintained
